atlas/AtlasMl/atlasml/ml/MLPipelines/PipelineWorkflows.py

```
@@ -0,0 +1,241 @@
+            - All changes are atomic - either all succeed or none are applied
+
+        Warning:
+            Ensure both text_id and competency_id exist in the database before calling
+            this function to avoid potential errors.
+        """
+
+        text = self.weaviate_client.get_embeddings_by_property(CollectionNames.TEXT.value, "text_id", text_id)
+        competency = self.weaviate_client.get_embeddings_by_property(CollectionNames.COMPETENCY.value, "competency_id", competency_id)
+        cluster = self.weaviate_client.get_embeddings_by_property(CollectionNames.CLUSTERCENTER.value, "cluster_id", competency["properties"]["cluster_id"])
+
+        new_text_competencyID = text["properties"]["competency_ids"]
```

Ruff (0.11.9)

Variable `new_text_competencyID` in function should be lowercase

(N806)